presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PushAggregationIntoTableScan.java

```diff
@@ -0,0 +1,214 @@
+
+        // The new scan outputs and assignments are union of existing and the ones returned by connector pushdown.
+        ImmutableList.Builder<Symbol> newScanOutputs = new ImmutableList.Builder<>();
+        newScanOutputs.addAll(tableScan.getOutputSymbols());
+
+        ImmutableMap.Builder<Symbol, ColumnHandle> newScanAssignments = new ImmutableMap.Builder<>();
+        newScanAssignments.putAll(tableScan.getAssignments());
+
+        Map<String, Symbol> variableMappings = new HashMap<>();
+
+        for (Assignment assignment : result.getAssignments()) {
+            Symbol symbol = context.getSymbolAllocator().newSymbol(assignment.getVariable(), assignment.getType());
+
+            newScanOutputs.add(symbol);
+            newScanAssignments.put(symbol, assignment.getColumn());
+            variableMappings.put(assignment.getVariable(), symbol);
+        }
+
+        List<Expression> newPartialProjections = result.getProjections().stream()
+                .map(expression -> ConnectorExpressionTranslator.translate(expression, variableMappings, new LiteralEncoder(metadata)))
+                .collect(toImmutableList());
+
+        Assignments.Builder assignmentBuilder = Assignments.builder();
+        IntStream.range(0, aggregationOutputSymbols.size())
+                .forEach(index -> assignmentBuilder.put(aggregationOutputSymbols.get(index), newPartialProjections.get(index)));
+
+        // projections assignmentBuilder should have both agg and group by so we add all the group bys as symbol references
+        groupingSets.getGroupingKeys().stream()
+                .forEach(groupBySymbol -> assignmentBuilder.put(groupBySymbol, groupBySymbol.toSymbolReference()));
+
+        return Result.ofPlanNode(
+                new ProjectNode(
+                        context.getIdAllocator().getNextId(),
+                        TableScanNode.newInstance(
+                                context.getIdAllocator().getNextId(),
+                                handle,
+                                newScanOutputs.build(),
+                                newScanAssignments.build()),
+                        assignmentBuilder.build()));
+    }
+
+    private List<AggregateFunction> toAggregateFunction(List<AggregationNode.Aggregation> aggregations)
+    {
+        ImmutableList.Builder<AggregateFunction> aggregateFunctionBuilder = ImmutableList.builder();
+        for (AggregationNode.Aggregation aggregation : aggregations) {
+            Signature signature = aggregation.getResolvedFunction().getSignature();
+            Type returnType = metadata.getType(signature.getReturnType());
+            String name = signature.getName();
+            List<Expression> arguments = aggregation.getArguments();
+            List<Symbol> symbols = arguments.stream()
+                    .filter(expression -> expression instanceof SymbolReference)
+                    .map(SymbolsExtractor::extractUnique)
+                    .flatMap(Collection::stream)
```

Comment:
This can only be a SymbolReference, so it'd be more appropriate to do:

```suggestion
                    .map(Symbol::from)
```

Reply:
Done.